README.md

```diff
@@ -44,46 +42,321 @@ see https://github.com/ioBroker/ioBroker/wiki/Installation
+        "sysLogType_comment": "The type of the syslog protocol to use (Default: BSD).",
+        "app_name_comment": "The name of the application (Default: process.title).",
+        "eol_comment": "The end of line character to be added to the end of the message (Default: Message without modifications)."
+      }
+    }
+  },
+```
+
+### Controlling and Monitoring of Adapter processes
+
+#### Multiple Adapter instances
+**Feature Status: Stable**
+
+ioBroker allows to install multiple adapters on the sytem. For each adapter multiple instances can be created and started with an independent configuration.
+
+#### Start Adapter instances as normal processes
+**Feature Status: Stable**
+
+By default adapters are started by the js-controller as separate nodejs processes. They connect to the State and Objects DBs to read and write their data and some statistical information.
+
+The js-controller is monitoring the process-id of the started process and also controls stopping or restarting these processes.
+
+With this approach the whole iobroker system is very robust. A faulty adapter will only affect his own process and the js-controller and the other adapter processes are unaffected.
+
+The downside of the approach is that more RAM is needed because each node.js process needs 20-30MB RAM for the Node.JS part of it. With this a 1GB system is usually limited to run approx 10 to max 15 adapter processes. Beside using systems with more RAM also splitting adapters on multiplte hosts in a multihost environment could be an idea.
+
+#### Start Adapter instances in compact mode
+**Feature Status: Technology Preview since 2.0.0**
+
+The compact mode is developed especially for low memory systems. All adapters configured to run in compact mode will run inside the js-controller process.
+
+The positive effect of this is that each adapter needs 20-30MB less RAM because it do not need to have it's own node.js process handling. With this also a 512MB RAM system can run several adapter instances.
+
+The downside is that as soon as one adapter is reacting faulty and generate an unhandled error, the whole js-controller process will be restarted.
+
+This means that this feature is especially good for slave systems but not for the ioBroker master process.
+
+To enable compact mode for an js-controller instance you need to manually change the iobroker.json configuration file:
+
+```
+{
+  "system": {
+    ...
+    "compact":true
+  },
+```
+
+Then restart the js-controller.
+All adapters that can run in compact mode will also be started in compact mode.
+
+##### Implementation details
+
+https://forum.iobroker.net/topic/18338/experimentell-js-controller-compact-mode
+
+##### Testing details for Development
+
+For testing setup your js.controller to use compact mode and change the io-package and javascript files accordingly. Do not forget to ```iobroker upload adaptername``` after changes to io-package.json.
+
+One important part is starting the adapter. 
```

Comment:
Do you mean restarting? Shouldn't that happen on upload anyways?

Reply:
One important part to check is that the adapter start works as expected also in compact mode.